app/services/ems_physical_infra_dashboard_service.rb

```diff
@@ -0,0 +1,42 @@
+class EmsPhysicalInfraDashboardService < DashboardService
+  include UiServiceMixin
+
+  def initialize(ems_id, controller)
+    @ems_id = ems_id
+    @ems = EmsPhysicalInfra.find(@ems_id) if @ems_id.present?
```

Comment:
Maybe it's totally wrong, but isn't this something that we should wrap with RBAC?

Reply:
tbh... I don't know @skateman, I haven't seen any example of a service wrapped with RBAC

Reply:
https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/services/topology_service.rb#L70
ping @lpichler

Reply:
Oh, so are you suggesting to wrap in RBAC the relations of a physical provider?

Reply:
@felipedf you wrap it:
`Rbac::Filterer.filtered(EmsPhysicalInfra.where(:id => @ems_id)) ` or
`Rbac::Filterer.filtered(EmsPhysicalInfra.find(@ems_id)) ` or

@felipedf I think that @skateman is thinking about this ^ or not ?  

> the relations of a physical provider
We are not able to do in one RBAC call, if we need it use on relation but if we need to wrap it - we have to do probably on places where these relations are listed.

Reply:
Like @felipedf said - the RBAC checks have already happened in the main controller - `EmsPhysicalInfraController` - so wouldn't doing it here again be redundant?

Reply:
so in that case yes it would be redundant :) and it is not needed, thanks @AparnaKarve ! 

and can you show me where is the RBAC check in main controller, to learn something from UI :)
thanks!

Reply:
@lpichler sure! :)


All of the Ems controllers have `Mixins::GenericShowMixin` included.
So if you check the `init_show` method in generic_show_mixin, you will see -
`@record = identify_record(params[:id], model_class)`

which is where the rbac check is --

```rb
def identify_record(id, klass = self.class.model)
  begin
    record = find_record_with_rbac(klass, id)
    rescue => @bang
    self.x_node = "root" if @explorer
    flash_to_session(@bang.message, :error, true)
end
   record
end
```

Reply:
Ok, that excludes maybe this line, but what about https://github.com/ManageIQ/manageiq-ui-classic/pull/3806/files#diff-56332bbfcf2fe957b5164fc8d340288fR34?

Reply:
OK, so... there is no RBAC check here and it indeed needs to be here.

`EmsPhysicalInfraDashboardService` gets the id as a param to the constructor...
It gets initialized only in one place .. 

```
app/controllers/ems_physical_infra_dashboard_controller.rb
22:    EmsPhysicalInfraDashboardService.new(ems_id, self).recent_servers_data
```

- that's the `recent_servers` method, only called from `recent_servers_data`:

```
  def recent_servers_data
    render :json => {:data => recent_servers(params[:id])}
  end
```


So... `ems_id` is really `params[:id]` and **definitely needs** an RBAC check :).

Reply:
To add a note - doing an RBAC check twice is a potential performance problem.
**Not** doing an RBAC check at all is a definite security problem.

Security trumps performance for the non-insane cases :)

Reply:
(Also, `EmsPhysicalInfraDashboardController` does not use that `Mixins::GenericShowMixin` at all.)

Reply:
Why does `EmsPhysicalInfraDashboardController` exist in the first place?
Because of the `EmsPhysicalInfraController::show` method.

So I would think all the `@ems` references in `EmsPhysicalInfraDashboardController`  are already vetted by the main controller `EmsPhysicalInfraDashboardController` for rbac -- at least, that would be the expected behavior.

Reply:
A note to @lpichler's comment:

> @felipedf you wrap it:
> Rbac::Filterer.filtered(EmsPhysicalInfra.where(:id => @ems_id)) or
> Rbac::Filterer.filtered(EmsPhysicalInfra.find(@ems_id)) or

There are methods to do this: https://github.com/ManageIQ/manageiq-ui-classic/blob/b1077efd9cad7ce8c0fd1acef3c233b5936ec2dc/app/controllers/mixins/checked_id_mixin.rb#L110-L112

more info in https://github.com/ManageIQ/guides/blob/master/reviewers_guidelines.md#testing-rbac-on-checked-items

Reply:
> Why does EmsPhysicalInfraDashboardController exist in the first place?
Because of the EmsPhysicalInfraController::show method.

The problem is, the request to `EmsPhysicalInfraController` is a different request than the one talking to `EmsPhysicalInfroDashboardController`.

So, you're right the first request does RBAC.

But that doesn't really matter because the only guarantee you have that the second will use the same id is if the user doesn't change it.

(And the point of security checks is to catch bad behaviour, not to rely on good one.)

Reply:
Although `params[:id]` does cause a security concern -- so for that reason, I would say we would need rbac.

Not sure if we can somehow avoid using `params[:id]` here